### PR TITLE
Sync all unstable features with Unstable Book; add tidy lint. 

### PR DESCRIPTION
**Pull request by frewsxcv**
Add a tidy lint that checks for...

* Unstable Book sections with no corresponding SUMMARY.md links
* unstable features that don't have Unstable Book sections
* Unstable Book sections that don't have corresponding unstable features

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Urg, that's a LOT of code! Fine by me otherwise.

**Comment by frewsxcv** [comment]
@GuillaumeGomez Which code are you talking about?

File: src/doc/unstable-book/src/field-init-shorthand.md [hunk 1]
```diff
@@ -1,10 +0,0 @@
-# `field_init_shorthand`
-
-The tracking issue for this feature is: [#37340]
-
-[#37340]: https://github.com/rust-lang/rust/issues/37340
-
-------------------------
```
Comment (steveklabnik) [inline src/doc/unstable-book/src/field-init-shorthand.md]: so, I purposefully put these at the end of sections that do not have docs, just to keep it consistent; would you mind updating the ones you've added to have this?

**Review by steveklabnik — DISMISSED** [review]
looks good to me, but one stylistic issue

**Comment by stjepang** [comment]
I'm confused by the deletion of `sort-unstable.md`.
This is a new feature, not a stabilized one - am I missing something?

**Comment by frewsxcv** [comment]
@stjepang `sort_unstable` hasn't landed in `master` yet, though I do see the open PR. If that one merges before this one, I'll remove the deletion.

**Comment by frewsxcv** [comment]
Oh, the implementation of the feature landed five days ago. Huh, I'll have to check why my tidy lint wanted it deleted. Good catch

**Comment by stjepang** [comment]
Actually, the PR for the feature was submitted 5 days ago, and it landed an hour ago.
Nothing wrong with tidy lint. :)

File: src/doc/unstable-book/src/field-init-shorthand.md (hunk 1 above)
Comment (frewsxcv) [inline src/doc/unstable-book/src/field-init-shorthand.md]: Addressed in the latest force push.

**Comment by frewsxcv** [comment]
Anyone else see anything? If everything looks good, just give me a "r=me" thing. I don't want to approve this until after https://github.com/rust-lang/rust/pull/40619 merges.

**Comment by frewsxcv** [comment]
https://github.com/rust-lang/rust/pull/40619 merged. This is ready to go now. r? @steveklabnik

**Comment by frewsxcv** [comment]
I have no idea what Travis is getting hung up on here. If anyone has thoughts, let me know.

**Comment by frewsxcv** [comment]
I don't want this to bit-rot while I try to debug this, so I extracted all the non-tidy-lint changes into a new PR: https://github.com/rust-lang/rust/pull/40786

**Comment by frewsxcv** [comment]
This should be ready to go now.

r? @steveklabnik

**Comment by steveklabnik** [comment]
@bors: r+

🎊

**Comment by frewsxcv** [comment]
bors wasn't detecting this pull request. Just amended the commit.

@bors r=steveklabnik

**Comment by frewsxcv** [comment]
@bors retry

timed out while trying to connect to github

**Comment by frewsxcv** [comment]
@bors r=steveklabnik p=1

new feature got added a few hours ago

going to give this a p=1 to make sure this stops getting out of sync

**Pull request merged**